tests/features/communities/eupl/jla.feature

```diff
@@ -66,3 +71,154 @@ Feature:
     Then I should see the text "2 licences found"
     And I should see the text "Foo Licence"
     And I should see the text "Bar Licence"
+
+  Scenario: Test the licence comparer.
+
+    Given SPDX licences:
+      | uri                         | title              | ID         |
+      | http://joinup.eu/spdx/baz   | SPDX licence baz   | SPDX_BAZ   |
+      | http://joinup.eu/spdx/qux   | SPDX licence qux   | SPDX_QUX   |
+      | http://joinup.eu/spdx/quux  | SPDX licence quux  | SPDX_QUUX  |
+      | http://joinup.eu/spdx/waldo | SPDX licence waldo | SPDX_WALDO |
+
+    # Test the page when the comparision list is missed.
+    When I am on "/licence/compare"
+    Then I should get a 404 HTTP response
+
+    # Test the page when there's only one licence.
+    When I am on "/licence/compare/SPDX_FOO"
+    Then I should get a 404 HTTP response
+
+    # Test the page when there are too many licences.
+    When I am on "/licence/compare/SPDX_FOO/SPDX_BAR/SPDX_BAZ/SPDX_QUX/SPDX_QUUX/SPDX_WALDO"
+    Then I should get a 404 HTTP response
+
+    # Test the page when there are invalid SPDX IDs licences.
+    When I am on "/licence/compare/ARBITRARY/SPDX/LICENCES"
+    Then I should get a 404 HTTP response
+
+    # Test the page with SPDX IDs without a corresponding Joinup licence.
+    When I am on "/licence/compare/SPDX_FOO/SPDX_BAR/SPDX_BAZ/SPDX_QUX/SPDX_QUUX"
+    Then I should get a 404 HTTP response
+
+    When I visit "/licence/compare/SPDX_FOO/SPDX_BAR"
```

Comment:
Changing in the tests the `SPDX_FOO` with `SPDX_FOO+2.51-3` breaks the test here.

Reply:
The IDs in route path is a mess and doesn't work. I switched to query string.